modules/tensor_mechanics/src/materials/CZM/CZM3DCLaw.C

```diff
@@ -0,0 +1,169 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "CZM3DCLaw.h"
+
+registerMooseObject("TensorMechanicsApp", CZM3DCLaw);
+
+template <>
+InputParameters
+validParams<CZM3DCLaw>()
+{
+  InputParameters params = validParams<CZMMaterialBase>();
+  params.addClassDescription("3DC cohseive law model, no damage");
+  params.addRequiredParam<Real>(
+      "normal_gap_at_maximum_normal_traction",
+      "the value of normal gap at which maximum normal traction is achieved");
+  params.addRequiredParam<Real>(
+      "tangential_gap_at_maximum_shear_traction",
+      "the value of tangential gap at which maximum shear traction is achieved");
+  params.addRequiredParam<Real>("maximum_normal_traction",
+                                "The maximum normal traction the interface can sustain");
+  params.addRequiredParam<Real>("maximum_shear_traction",
+                                "The maximum shear traction the interface can sustain");
+  params.addClassDescription("Simple Exponential cohseive law model, with damage");
```

Comment:
Delete this line -- you should only have one class description.

Reply:
fair enough I'll delete it.

Reply:
done